solver/src/driver.rs

```diff
@@ -414,6 +421,14 @@ impl Driver {
 
             tracing::debug!("winning settlement: {:?}", settlement);
             self.submit_settlement(settlement.clone()).await;
+            self.inflight_orders = settlement
```

Comment:
Not entirely sure and no strong feeling, but maybe it would be less "leaky" to store the full trades in a hashset?

Reply:
Yes, although I might propose `HashMap<OrderUid, Trade>` so that we can easily search for `order_id` containment when flitering. Would this be a reasonable compromise?

Reply:
Given that we aren't actually handling this at the moment, I suppose we could get away without the `HashMap` for now.

Reply:
I would argue for a `HashSet<OrderUid>` since we don't really use the executed amount for now.

Reply:
I'm not fond of `HashSet<Trade>` because trade equality is brittle, and, at least for me, adds a lot of ambiguity to the code without looking around to fully understand the context. For example, imagine an order:
```rust
let order = Order { partially_fillable: true, ... };
let mut trades = HashSet::new();

trades.insert(Trade {
    order: order.clone(),
    sell_token_index: 0,
    buy_token_index: 1,
    executed_amount: 100.into(),
});
// What do you expect to happen here? Its the same order, so should it replace the existing trade in the
// hash set? The token indices are different, what does this mean and is it even possible here?
trades.insert(Trade {
    order: order.clone(),
    sell_token_index: 1,
    buy_token_index: 0,
    executed_amount: 50.into(),
});
```

Reply:
The other issue I see with `HashMap<OrderUid, Trade>` is the ambiguity around trading a a partially fillable order multiple times in the same settlement. In that case `HashMap<OrderUid, Trade>` would not be enough as only one of the trades would be recorded.

I would either suggest `Vec<Trade>` and with the `.map().collect()` as you do here, or simply just use `HashSet<OrderUid>` for now, until we actually need trade data when filtering.